docs/csharp/whats-new/csharp-9.md

```diff
@@ -0,0 +1,233 @@
+
+## Pattern matching enhancements
+
+C# 9 includes new pattern matching improvements:
+
+- ***Type patterns*** match a variable is a type
+- ***Parenthesized patterns*** enforce or emphasize the precedence of pattern combinations
+- ***Conjunctive `and` patterns*** require both patterns to match
+- ***Disjunctive `or` patterns*** require either pattern to match
+- ***Negated `not` patterns*** require that a pattern doesn’t match
+- ***Relational patterns*** require the input be less than, greater than, less than or equal, or greater than or equal to a given constant.
+
+These patterns enrich the syntax for patterns. Consider these examples:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/PatternUtilities.cs" ID="IsLetterPattern":::
+
+Alternatively, with optional parentheses to make it clear that and has higher precedence than or:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/PatternUtilities.cs" ID="IsLetterOrSeparatorPattern":::
+
+One of the most common uses is a new syntax for a null check:
+
+```csharp
+if (e is not null)
+{
+    // ...
+}
+```
+
+Any of these patterns can be used in any context where patterns are allowed: `is` pattern expressions, `switch` expressions, nested patterns, and the pattern of a `switch` statement’s `case` label.
+
+## Performance and interop
+
+Three new features improve support for native interop and low-level libraries that require high performance: native sized integers, function pointers, and omitting the `localsinit` flag.
+
+Native sized integers, `nint` and `nuint`, are integer types. They are expressed by the underlying types <xref:System.IntPtr?displayProperty=nameWithType> and <xref:System.UIntPtr?displayProperty=nameWithType>. The compiler surfaces additional conversions and operations for these types as native ints. Native sized ints do not have constants for `MaxValue` or `MinValue`, except for `nuint.MinValue`, which has a `MinValue` of `0`. Other values cannot be expressed as constants because it would depend on the native size of an integer on the target machine. You can use constant values for `nint` in the range [`int.MinValue` .. `int.MaxValue`]. You can use constant values for `nuint` in the range [u`int.MinValue` .. `uint.MaxValue`]. The compiler performs constant folding for all unary and binary operators using the <xref:System.Int32?displayProperty=nameWithType> and <xref:System.UInt32?displayProperty=nameWithType> types. If the result does not fit in 32 bits, the operation is executed at runtime and is not considered a constant. Native sized integers can increase performance in scenarios where integer math is used extensively and needs to have the faster performance possible.
```

Comment:
```suggestion
Native sized integers, `nint` and `nuint`, are integer types. They are expressed by the underlying types <xref:System.IntPtr?displayProperty=nameWithType> and <xref:System.UIntPtr?displayProperty=nameWithType>. The compiler surfaces additional conversions and operations for these types as native ints. Native sized ints do not have constants for `MaxValue` or `MinValue`, except for `nuint.MinValue`, which has a `MinValue` of `0`. Other values cannot be expressed as constants because it would depend on the native size of an integer on the target machine. You can use constant values for `nint` in the range [`int.MinValue` .. `int.MaxValue`]. You can use constant values for `nuint` in the range [`uint.MinValue` .. `uint.MaxValue`]. The compiler performs constant folding for all unary and binary operators using the <xref:System.Int32?displayProperty=nameWithType> and <xref:System.UInt32?displayProperty=nameWithType> types. If the result does not fit in 32 bits, the operation is executed at runtime and is not considered a constant. Native sized integers can increase performance in scenarios where integer math is used extensively and needs to have the fastest performance possible.
```

One of the `unit.MinValue` was incorrectly typed as "u`int.MinValue`". Also, fixed "faster", corrected as "fastest".